dagger-core/src/main/java/io/odpf/dagger/core/processors/external/http/request/HttpPostRequestHandler.java

```diff
@@ -39,7 +42,7 @@ public BoundRequestBuilder create() {
         String endpoint = httpSourceConfig.getEndpoint();
 
         if (!StringUtil.isNullOrEmpty(httpSourceConfig.getEndpointVariables())) {
```

Comment:
You need not do this validation check anymore as `endpointVariablesValues` cannot be null but can be empty.
You can just do 
`String endpoint = String.format(httpSourceConfig.getEndpoint(), endpointVariablesValues)` 

at line 42.